### ISSUE TITLE
CORS issue with cached info.json

### ISSUE THREAD

**Issue by mikeapp** [medusa-project/cantaloupe#107]
Using two clients on separate domains in the same browser instance leads to CORS issues if caching is turned on, due to the browser relying on the cached copy of the info.json.

I believe the Vary header should include Origin per https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS#Access-Control-Allow-Origin

**Comment by devanonyme** [comment]
We have a similar issue in our local and dev environnment using the same instance of Cantaloupe.
If I get the images from our dev site, the image gets cached with CORS from dev site. If go to the same images using my local runing site, the images are then fetched from browser cache with uncompatible cache headers.

We had quite a few CORS issues to handle (support local testing with mobile devices) and I have decided to just handle it manually with a clear cache.

**Comment by adolski** [comment]
This is unfortunately blocked by https://github.com/restlet/restlet-framework-java/issues/187

**Comment by devanonyme** [comment]
Not too much of a problem for us as the problem occur only in our testing environnement and we can probably add it in our Apache Conf
Header merge Vary Origin

I will check with my coworkers and report if this solution works as expected.

**Comment by imoutsatsos** [comment]
I'm experiencing the same issue in a DEV/PROD environment using the same Cantaloupe server similar to @devanonyme How can I manually clear the cache?

**Comment by adolski** [comment]
> I'm experiencing the same issue in a DEV/PROD environment using the same Cantaloupe server similar to @devanonyme How can I manually clear the cache?

See here: https://medusa-project.github.io/cantaloupe/manual/3.3/caching.html#Maintenance

**Comment by imoutsatsos** [comment]
thank you @adolski ! I also found that in my case, the browser cache seemed to be contributing to CORS errors. When I clear the browser image cache, the images load. I'm working with two Cantaloupe servers on different Windows servers. One image server serves raw images while the other serves normalized intensity images. Depending on the web client settings the user can select what type of images they will request and that sets the server prefix. I think that is the cause of my issues but since I can't use an iiif API call to get one or the other type of images if the same image is requested raw and then intensity normalized the client cache detects different Access control header and complains with:
```
XMLHttpRequest cannot load http://blah1:8182/iiif/2/%2F%2FCPC%2FUSCA-YOKOGA…0170216_150044%2FNPC1_P2%2FNPC1_P2_M11_T0001F002L01A01Z01C01.tif/info.json. The 'Access-Control-Allow-Origin' header has a value 'http://blah2:8080' that is not equal to the supplied origin. Origin 'http://blah3:8080' is therefore not allowed access.
```

**Comment by devanonyme** [comment]
Hello, just got back from vacation.

Not sure to understand your setup but my problem really occured in such configuration 

http://blah1:8182 (cantaloup)
http://blah2:8080 (app server 1)
http://blah3:8080 (app server 2)

The browser keep in cache the ressources from blah1:8182 that contains origin header, so the first access will block the other. The only way to correct this is the use of the vary header.

Not a problem for us as this configuration exist in our dev environment only.

Another option could be to use a param based on the app server in use, that way the browser cache would be caching the ressource separatly, just partially bypassing the benefits of browser caching.

Ex.
/info.json?env=appserver1
/info.json?env=appserver2

**Comment by adolski** [comment]
This can get fixed as soon as Restlet merges [this pull request](https://github.com/restlet/restlet-framework-java/pull/1268) into a release.

**Comment by adolski** [comment]
`Origin` is now present in all `Vary` headers. This will be included in version 3.3.3. Thanks all.

**Comment by devanonyme** [comment]
Nice! Thanks!
